src/Features/CSharp/Portable/SignatureHelp/PrimaryConstructorBaseTypeSignatureHelpProvider.cs

```diff
@@ -0,0 +1,154 @@
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.SignatureHelp
+{
+    /// <summary>
+    /// Implements SignatureHelp and ParameterInfo for <see cref="PrimaryConstructorBaseTypeSyntax"/>
+    /// such as 'record Student(int Id) : Person($$"first", "last");`.
+    /// </summary>
+    [ExportSignatureHelpProvider("PrimaryConstructorBaseTypeSignatureHelpProvider", LanguageNames.CSharp), Shared]
+    internal partial class PrimaryConstructorBaseTypeSignatureHelpProvider : AbstractCSharpSignatureHelpProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public PrimaryConstructorBaseTypeSignatureHelpProvider()
+        {
+        }
+
+        public override bool IsTriggerCharacter(char ch)
+            => ch == '(' || ch == ',';
+
+        public override bool IsRetriggerCharacter(char ch)
+            => ch == ')';
+
+        private bool TryGetBaseTypeSyntax(SyntaxNode root, int position, ISyntaxFactsService syntaxFacts, SignatureHelpTriggerReason triggerReason, CancellationToken cancellationToken, out PrimaryConstructorBaseTypeSyntax expression)
+        {
+            if (!CommonSignatureHelpUtilities.TryGetSyntax(root, position, syntaxFacts, triggerReason, IsTriggerToken, IsArgumentListToken, cancellationToken, out expression))
+            {
+                return false;
+            }
+
+            return expression.ArgumentList != null;
+
+            static bool IsArgumentListToken(PrimaryConstructorBaseTypeSyntax expression, SyntaxToken token)
+            {
+                return expression.ArgumentList != null &&
+                    expression.ArgumentList.Span.Contains(token.SpanStart) &&
+                    token != expression.ArgumentList.CloseParenToken;
+            }
+        }
+
+        private bool IsTriggerToken(SyntaxToken token)
+            => SignatureHelpUtilities.IsTriggerParenOrComma<PrimaryConstructorBaseTypeSyntax>(token, IsTriggerCharacter);
+
+        protected override async Task<SignatureHelpItems?> GetItemsWorkerAsync(Document document, int position, SignatureHelpTriggerInfo triggerInfo, CancellationToken cancellationToken)
+        {
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+            if (root is null || syntaxFacts is null ||
+                !TryGetBaseTypeSyntax(root, position, syntaxFacts, triggerInfo.TriggerReason, cancellationToken, out var baseTypeSyntax))
+            {
+                return null;
+            }
+
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var within = semanticModel.GetEnclosingNamedTypeOrAssembly(position, cancellationToken);
+            var baseType = (INamedTypeSymbol?)semanticModel.GetTypeInfo(baseTypeSyntax.Type, cancellationToken).Type;
```

Comment:
any concern this won't be an INAmedTypeSymbol?  probably better to `as` cast this for error scenarios.

Reply:
actually... i'm fairly certain this can crash.  i.e. if you do  `record Foo<T> : T`